components/camel-google-calendar/src/main/java/org/apache/camel/component/google/calendar/stream/GoogleCalendarStreamConsumer.java

```diff
@@ -59,19 +70,21 @@ public GoogleCalendarStreamEndpoint getEndpoint() {
 
     @Override
     protected int poll() throws Exception {
-        com.google.api.services.calendar.Calendar.Events.List request
-                = getClient().events().list(getConfiguration().getCalendarId()).setOrderBy("updated");
+        Calendar.Events.List request = getClient().events().list(getConfiguration().getCalendarId());
         if (ObjectHelper.isNotEmpty(getConfiguration().getQuery())) {
```

Comment:
Please don't use this class, we have all the tools to check this stuff the camel way

Reply:
I guess this refers to `Preconditions`? What is the Camel way? Could you point me in the right direction?

Reply:
Use ObjectHelper class, check if the syncFlow is not empty and log a warning, I don't think we have to throw an exception.

Reply:
I was referring to https://github.com/apache/camel/blob/master/components/camel-google-calendar/src/main/java/org/apache/camel/component/google/calendar/GoogleCalendarEndpoint.java#L66 where an exception is thrown when incompatible option is set. Invalid configuration results in request failures, so it's better to fail fast. 

Moreover, `syncFlow` may not be empty, it just shouldn't be set to `true`.

Reply:
Then instead of doing this check here in the poll, check if syncFlow is true and query is set in the endpoint consumer creation and fail there. Don't use Precondition, is just another class that will change in the future and we'll have to align, you just need to check a boolean value.

Reply:
Done.